lib/IRGen/GenValueWitness.cpp

```diff
@@ -1380,17 +1380,24 @@ static void addValueWitnessesForAbstractType(IRGenModule &IGM,
                                              CanType abstractType,
                                              bool &canBeConstant) {
   std::optional<BoundGenericTypeCharacteristics> boundGenericCharacteristics;
-  if (auto boundGenericType = dyn_cast<BoundGenericType>(abstractType)) {
-    CanType concreteFormalType = getFormalTypeInPrimaryContext(abstractType);
-
-    auto concreteLoweredType = IGM.getLoweredType(concreteFormalType);
-    const auto *boundConcreteTI = &IGM.getTypeInfo(concreteLoweredType);
-    auto packing = boundConcreteTI->getFixedPacking(IGM);
-    boundGenericCharacteristics = {concreteLoweredType, boundConcreteTI,
-                                   packing};
-
-    abstractType =
-        boundGenericType->getDecl()->getDeclaredType()->getCanonicalType();
+  // Embedded existentials need fully "specialized" value witness table
+  // functions: i.e the metadata argument must remain unused.
+  // For the non-embedded existentials path, when there is a bound generic type,
+  // that is we have a specialized generic type, we have decided for code size
+  // reasons to continue using "generic" value witness table functions i.e the
+  // same once used for runtime instantiated generic metadata.
+  if (!IGM.Context.LangOpts.hasFeature(Feature::EmbeddedExistentials)) {
+    if (auto boundGenericType = dyn_cast<BoundGenericType>(abstractType)) {
```

Comment:
What is this code used for? It's already wrong if you have a non-generic type nested in a generic one, like
```
struct G<T> {
  struct Inner {}
}
```

Something like `G<Int>.Inner` is not a BGT